src/shared/sleep-config.c

```diff
@@ -143,6 +146,136 @@ int read_battery_capacity_percentage(void) {
         return battery_capacity;
 }
 
+/*Read system and battery identifier from specific location and generate hash of it*/
+static int get_system_battery_identifier_hash(uint8_t ret[static SHA256_DIGEST_SIZE]) {
+        _cleanup_free_ char *bat_manufacturer = NULL, *bat_model_name = NULL, *bat_serial_number = NULL;
+        sd_id128_t machine_id, product_id;
+        struct sha256_ctx hash;
+        int r;
+
+        sha256_init_ctx(&hash);
+        r = read_one_line_file("/sys/class/power_supply/BAT0/manufacturer", &bat_manufacturer);
+        if (r > 0)
+               sha256_process_bytes(bat_manufacturer, strlen(bat_manufacturer), &hash);
+        else if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/manufacturer is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/manufacturer: %m");
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/model_name", &bat_model_name);
+        if (r > 0)
+               sha256_process_bytes(bat_model_name, strlen(bat_model_name), &hash);
+        else if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/model_name is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/model_name: %m");
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/serial_number", &bat_serial_number);
+        if (r > 0)
+               sha256_process_bytes(bat_serial_number, strlen(bat_serial_number), &hash);
+        else if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/serial_number is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/serial_number: %m");
+
+        r = sd_id128_get_machine(&machine_id);
+        if (r > 0)
```

Comment:
this and the one below return 0 on success, so machine-id and product-id will never be added to the hash

instead of checking it first, just do the process_bytes unconditionally after each error check. Same for the strings above, unless an error was returned, the return buffer will be valid (might be empty but that's fine)

Reply:
check the implementation, it's returning 0 on success, which means it is not used here